### PR TITLE
Added auto-formatting, avro-specific code style settings, and some parsing corner cases

### PR DESCRIPTION
**Pull request by mwilbz**
**Formatting**

Pretty basic. Just does indentations, doesn't handle wrapping or spacing or any of the other fancy stuff.

**Code Style Settings**

Ability to have avdl-specific indentation levels

**Parsing/Lexing Changes**

- Allow multiple annotations before a protocol
- Allow "-" characters in annotations, keep it from being highlighted as a bad character
- Allow array of array of strings to be argument of an @annotation

### CODE REVIEW

**Comment by AbigailBuccaneer** [comment]
Brilliant, thanks for taking the time to do all this! I'll give the code a proper read-through tomorrow.

**Comment by mwilbz** [comment]
@AbigailBuccaneer have you had a chance to look at the code? Thank you for reviewing! =)

File: src/claims/bold/intellij/avro/idl/AvroIdl.bnf [hunk 1]
```diff
@@ -70,11 +70,12 @@
 
 file ::= protocol_def*
 
-protocol_def ::= annotation? 'protocol' IDENTIFIER '{' declaration* '}' { pin = 2 }
+protocol_def ::= annotation* 'protocol' IDENTIFIER '{' declaration* '}' { pin = 2 }
 
 annotation ::= '@' IDENTIFIER '(' annotation_body ')' { pin = 1 }
```
Comment (AbigailBuccaneer) [inline src/claims/bold/intellij/avro/idl/AvroIdl.bnf:75]: It looks like `annotation_body` should probably be replaced with `json_value` (see https://github.com/apache/avro/blob/c22a1a102d78558e04249dad2cc086ca647fc2dd/lang/java/compiler/src/test/idl/input/simple.avdl#L37)

**Review by AbigailBuccaneer — APPROVED** [review]
Thanks again! Looking over this PR has caused me to spot some issues elsewhere. I'll try and get a new version out soon.

File: src/claims/bold/intellij/avro/idl/AvroIdl.bnf (hunk 1 above)
Comment (mwilbz) [reply src/claims/bold/intellij/avro/idl/AvroIdl.bnf:75]: Agreed, good callout. I should have investigated what the more general possibilities were. Thanks for digging deeper!

**Comment by AbigailBuccaneer** [comment]
Released v0.3.0, uploaded to the JetBrains plugin repository, waiting approval. Hooray!

**Pull request merged**